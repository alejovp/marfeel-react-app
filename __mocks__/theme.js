export const themeMock = {
    tipography: {
        primaryFontColor: 'black',
        fontFamily: 'Helvetica, sans-serif'
    },
    header: {
        padding: 15
    },
    sectionMenu: {
        color: 'orange',
        textSize: 1.1 //em
    },
    background: {
        color: 'gray'
    },
    burgerButton: {
        color: 'red'
    }
};
