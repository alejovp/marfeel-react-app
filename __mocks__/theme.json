{
    "tipography": {
        "primaryFontColor": "black",
        "fontFamily": "Helvetica, sans-serif"
    },
    "header": {
        "padding": 15
    },
    "sectionMenu": {
        "color": "orange",
        "textSize": 1.1
    },
    "background": {
        "color": "gray"
    },
    "burgerButton": {
        "color": "red"
    },
    "logo": {
        "image": "https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg",
        "size": 36
    }
}