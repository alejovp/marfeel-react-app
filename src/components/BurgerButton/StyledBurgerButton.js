import styled from 'styled-components';
import { BURGER_BUTTON_SIZE } from '../../internals/constants';


export const StyledBurgerButton = styled.button`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: ${BURGER_BUTTON_SIZE}px;
    height: ${BURGER_BUTTON_SIZE}px;
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0;
    
    &:focus {
        outline: none;
    }
`;

StyledBurgerButton.displayName = 'StyledBurgerButton';
