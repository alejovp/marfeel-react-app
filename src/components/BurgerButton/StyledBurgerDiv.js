import styled from 'styled-components';
import { BURGER_BUTTON_SIZE } from '../../internals/constants';

export const StyledBurgerDiv = styled.div`
    width: ${BURGER_BUTTON_SIZE}px;
    height: 4px;
    background: ${({ theme }) => theme.burgerButton && theme.burgerButton.color || 'black'};
    border-radius: 10px;
    position: relative;
    transform-origin: 1px;
`;

StyledBurgerDiv.displayName = 'StyledBurgerDiv';
