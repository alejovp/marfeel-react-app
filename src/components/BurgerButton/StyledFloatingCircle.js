import styled from 'styled-components';
import { FLOATING_CIRCLE_PADDING } from '../../internals/constants';


export const StyledFloatingCircle = styled.div`
    position: absolute;
    left: ${({ left }) => left || 0}px;
    top: ${({ top }) => top || 0}px;
    padding: ${FLOATING_CIRCLE_PADDING}px;
    background-color: ${({theme}) => theme.background && theme.background.color || 'white'};
    border-radius: 50%;
    z-index: 10;
`;

StyledFloatingCircle.displayName = 'StyledFloatingCircle';
