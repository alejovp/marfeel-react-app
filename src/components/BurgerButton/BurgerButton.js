import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { StyledBurgerButton } from './StyledBurgerButton';
import { StyledBurgerDiv } from './StyledBurgerDiv';
import { StyledFloatingCircle } from './StyledFloatingCircle';
import { ThemeContext } from 'styled-components';
import {
    APP_BAR_HEIGHT,
    BURGER_BUTTON_SIZE,
    FLOATING_CIRCLE_PADDING
} from '../../internals/constants';


export const BurgerButton = ({ onClick }) => {
    const { header } = useContext(ThemeContext);

    return (
        <StyledFloatingCircle
            top={((APP_BAR_HEIGHT - BURGER_BUTTON_SIZE) / 2) - FLOATING_CIRCLE_PADDING}
            left={header && header.padding}
        >
            <StyledBurgerButton onClick={onClick}>
                <StyledBurgerDiv />
                <StyledBurgerDiv />
                <StyledBurgerDiv />
            </StyledBurgerButton>
        </StyledFloatingCircle>
    );
};

BurgerButton.propTypes = {
    onClick: PropTypes.func
};
