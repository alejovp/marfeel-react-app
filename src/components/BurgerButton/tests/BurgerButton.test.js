import React from 'react';
import { BurgerButton } from '../BurgerButton';
import theme from '../../../../__mocks__/theme.json';
import { mountWithTheme } from '../../../../scripts/jest/setup';
import { StyledBurgerButton } from '../StyledBurgerButton';
import { StyledBurgerDiv } from '../StyledBurgerDiv';
import { StyledFloatingCircle } from '../StyledFloatingCircle';


describe('<BurgerButton />', () => {

    describe('shape', () => {
        const renderedComponent = mountWithTheme(<BurgerButton />, theme);

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a StyledFloatingCircle component', () => {
            expect(renderedComponent.find(StyledFloatingCircle)).toHaveLength(1);
        });

        it('will render a StyledBurgerButton component', () => {
            expect(renderedComponent.find(StyledBurgerButton)).toHaveLength(1);
        });

        it('will render 3 StyledBurgerDiv elems', () => {
            expect(renderedComponent.find(StyledBurgerDiv)).toHaveLength(3);
        });

    });

    describe('props', () => {

        it('will invoke the onClick callback on click event', () => {
            const onClickMock = jest.fn();
            const renderedComponent = mountWithTheme(
                <BurgerButton onClick={onClickMock} />, theme);
            const button = renderedComponent.find(StyledBurgerButton);
            button.simulate('click');
            expect(onClickMock).toBeCalledTimes(1);
        });

    });

});
