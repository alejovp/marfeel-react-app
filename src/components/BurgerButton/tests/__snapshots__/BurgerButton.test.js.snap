// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerButton /> shape will render correctly 1`] = `
<BurgerButton>
  <StyledFloatingCircle
    left={15}
    top={8}
  >
    <div
      className="GCPYb"
    >
      <StyledBurgerButton>
        <button
          className="hXkIwj"
        >
          <StyledBurgerDiv>
            <div
              className="desPIF"
            />
          </StyledBurgerDiv>
          <StyledBurgerDiv>
            <div
              className="desPIF"
            />
          </StyledBurgerDiv>
          <StyledBurgerDiv>
            <div
              className="desPIF"
            />
          </StyledBurgerDiv>
        </button>
      </StyledBurgerButton>
    </div>
  </StyledFloatingCircle>
</BurgerButton>
`;
